### PR TITLE
[quant][pt2] Handle no conv bias in prepare QAT fusion

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100941
* __->__ #100610

Summary: This commit adds support for conv + BN fusion for the
case where conv has no bias. Since the replacement patterns with
and without conv bias are substantially different, we perform the
replacement for each of these two cases separately.

Test Plan:
python test/test_quantization.py TestQuantizePT2E.test_prepare_qat_conv_bn_fusion_no_conv_bias

Reviewers: jerryzh168, kimishpatel

Differential Revision: [D45743510](https://our.internmc.facebook.com/intern/diff/D45743510)

### CODE REVIEW

File: torch/ao/quantization/_pt2e/qat_utils.py [hunk 1]
```diff
@@ -94,11 +138,43 @@ def _fuse_conv_bn_qat(m: GraphModule) -> GraphModule:
...
+    # Here we do replacement separately for cases with and without conv bias, since
+    # the replacement patterns for these two cases are substantially different.
     # TODO: use the public replace_pattern API once it also returns replacement nodes
-    match_and_replacement = _replace_pattern(m, match_pattern, replacement_pattern, ignore_literals=True)
+
+    replacement_pattern_with_conv_bias = _get_aten_graph_module(
+        _fused_qat_conv2d_bn_pattern,
+        example_inputs,
+    )
+    mr_with_conv_bias = _replace_pattern(
```
Comment (jerryzh168) [inline torch/ao/quantization/_pt2e/qat_utils.py:152]: nit: we can use `replace_pattern_with_match_filters` now

**Review by jerryzh168 — APPROVED** [review]
LGTM thanks

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45743510).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45743510).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45743510).

**Pull request closed without merging**